app/assets/javascripts/discourse/lib/discourse-location.js.es6

```diff
@@ -101,7 +101,7 @@ const DiscourseLocation = Ember.Object.extend({
     const state = this.getState();
     path = this.formatURL(path);
 
-    if (state && state.path !== path) {
+    if (!state || (state && state.path !== path)) {
```

Comment:
Shouldn't this be?

```suggestion
    if (!state || state.path !== path) {
```

Reply:
yes kinda... I think it’s less prone to error in the future to have two clear conditions, instead of relying on the first condition to validate the second...

But I can see why you wouldn’t agreee 😅

Reply:
Up to you, both are fine to me

Reply:
I think @ZogStriP 's is more clear, let's go with that.